controllers/apim/ratelimitpolicy_status.go

```diff
@@ -85,3 +99,60 @@ func (r *RateLimitPolicyReconciler) availableCondition(specErr error) *metav1.Co
 
 	return cond
 }
+
+// gatewaysRateLimits returns all gateway level rate limits configuration from all the
+// gateways where this rate limit policy adds configuration
+func (r *RateLimitPolicyReconciler) gatewaysRateLimits(ctx context.Context, rlp *apimv1alpha1.RateLimitPolicy) ([]apimv1alpha1.GatewayRateLimits, error) {
+	logger, _ := logr.FromContext(ctx)
+	gwKeys, err := r.rlpGatewayKeys(ctx, rlp)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			gwKeys = make([]client.ObjectKey, 0)
+		} else {
+			return nil, err
+		}
+	}
+
+	result := make([]apimv1alpha1.GatewayRateLimits, 0)
+
+	for _, gwKey := range gwKeys {
+		gw := &gatewayapiv1alpha2.Gateway{}
+		err := r.Client().Get(ctx, gwKey, gw)
+		logger.V(1).Info("get gateway", "key", gwKey, "err", err)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				continue
+			}
+			return nil, err
+		}
+
+		gwAnnotations := gw.GetAnnotations()
+		if gwAnnotations == nil {
+			gwAnnotations = map[string]string{}
```

Comment:
If there are no annotations, maybe 

```go
return result, nil // empty
```

Reply:
it could be `continue`, otherwise the method may miss other gateway rate limits.

Changing it.